src/frontend/src/lib/components/tokens/TokenExchangeBalance.svelte

```diff
@@ -1,21 +1,21 @@
 <script lang="ts">
 	import { isNullish, nonNullish } from '@dfinity/utils';
+	import { i18n } from '$lib/stores/i18n.store';
 	import type { TokenFinancialData } from '$lib/types/token';
 	import { formatUSD } from '$lib/utils/format.utils';
 
 	export let balance: TokenFinancialData['balance'];
 	export let usdBalance: TokenFinancialData['usdBalance'];
+	export let nullishBalanceMessage = '-';
 </script>
 
 <output class="break-all">
 	{#if nonNullish(balance) && nonNullish(usdBalance)}
 		{formatUSD({ value: usdBalance })}
+	{:else if isNullish(balance)}
+		<span class:animate-pulse={isNullish(balance)}>{nullishBalanceMessage}</span>
```

Comment:
you don't need the condition here anymore, right? it will always be true

Reply:
No this is still needed. The `if` above checks if the `balance` AND `usdBalance` is non nullish. In case that the `balance` is defined but the `usdbalance` is nullish this check also failes and the third option will take place.

Reply:
but `animate-pulse={isNullish(balance)}` will always be true, since it's inside `if isNullish(balance)}`